src/file/reader.rs

```diff
@@ -203,6 +205,39 @@ impl FileReader for SerializedFileReader {
   }
 }
 
+impl TryFrom<File> for SerializedFileReader {
```

Comment:
can we have a few simple tests for these?

Reply:
Yes, sure. I will add them.